contrib/fonttools/acorn2sfd.c

```diff
@@ -330,7 +333,7 @@ static void ReadChunk(FILE *file,struct Outlines *outline,int chunk) {
 	flag = r_getint(file);
     else if ( outline->version==6 )
 	flag = (1<<7);
-    if ( !flag&0x80000000 )
+    if ( (!flag)&0x80000000 )
```

Comment:
This change makes no sense, you're effectively ANDing a 1 or a 0 against that bitflag now.

Reply:
But this change literally makes no sense. When you do `(!flag)`, you're making that value either exactly 1 or 0. So when you AND it with `0x80000000`, you'll always get 0. 

You either want !(flag&0x80000000) or *maybe* `(~flag)&0x80000000`.

Reply:
yes, better choice == (~flag)&0x80000000, avoids the possible ! error

Reply:
Do you have a spec reference or something, because that's fairly different semantics to what it was doing before.

Reply:
riscosfonts.zip